### DISCUSSION TITLE
Question about "secure context" and self-signed certificate

### DISCUSSION THREAD

**Discussion by regunakyle** [dani-garcia/vaultwarden#4738] [Q&A] [answered]
(Feel free to close this question if this is off topic, I ask here since I think the maintainers are probably experts in this)

Hi, I am evaluating vaultwarden and is reading the docs. The [Enable HTTPS](https://github.com/dani-garcia/vaultwarden/wiki/Enabling-HTTPS) section mentions that:

> For proper operation of vaultwarden, enabling [HTTPS](https://en.wikipedia.org/wiki/HTTPS) is pretty much required nowadays, since the Bitwarden web vault uses [web crypto APIs](https://developer.mozilla.org/en-US/docs/Web/API/SubtleCrypto) that most browsers only make available in HTTPS contexts.

Then I read the MSN article about _secure context_ ([here](https://developer.mozilla.org/en-US/docs/Web/Security/Secure_Contexts)) and skimmed through the W3C specs ([here](https://w3c.github.io/webappsec-secure-contexts/)). Seems like HTTPS (regardless of validity of the certificate) is enough to establish a _secure context_. This means I can create a self-signed SSL certificate and use it with vaultwarden.

Is my reasoning correct? If it is wrong, could you point me to a relevant documentation about this?

Thanks.

**Comment by BlackDex** [comment, marked as answer]
It is correct. Only, some clients like the mobile clients might be hard to get the self signed cert to be accepted.

**Comment by regunakyle** [comment]
> It is correct. Only, some clients like the mobile clients might be hard to get the self signed cert to be accepted.

I see. Is this is caused by iOS/Android policy or the mobile client itself?

**Reply by BlackDex** [reply]
Not sure actually. Might be both hehe.

**Reply by regunakyle** [reply]
> Not sure actually. Might be both hehe.

Do the client on Windows/Linux/Mac also exhibit this behavior?

**Reply by BlackDex** [reply]
Not as far as i know. But i haven't tested it in a while actually

**Reply by regunakyle** [reply]
> Not as far as i know. But i haven't tested it in a while actually

Thanks for your answer. I just got curious, will get a cert from letsencrypt either way

**Reply by BlackDex** [reply]
That will work without any issues.
